libraries/autoload.rb

```diff
@@ -1,3 +1,7 @@
+unless Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
+  raise "The compat_resources cookbook does not support chef versions older than Chef 12.0.0"
```

Comment:
So it will raise an error anyway - gotcha.  While its a shame we can't maintain backwards compatibility, such is the price of progress, I suppose. :)